src/urban_disco/core.clj

```diff
@@ -1,14 +1,18 @@
 (ns urban-disco.core
   (:gen-class))
 
-(defn build-tile [x y] 
+(defn build-tile [x y grid-height] 
   {:x        x
    :y        y
-   :explored false})
+   :explored false
+   :up (if (= y 0) :wall :door)
+   :down (if (= (+ y 1) grid-height) :wall :door)
+   :left (if (= x 0) :wall :door)
+   :right (if (= (+ x 1) grid-height) :wall :door)})
 
 (defn build-row [width row-number]
   (let [row (range width)]
-    (into [] (map #(build-tile % row-number) row))))
+    (into [] (map #(build-tile % row-number width) row))))
```

Comment:
Comments pull from first 2 PRs:

1. Generating the grid here in build-row and build-grid might be a little creative using map.

I'd like to some thoughts on how I did this, and if there is a better way of creating a dynamic vector of vectors.

2. `range` returns a list instead of a vector, you cannot use update-in/assoc-in on a list.

There's still probably a better way to create this.